src/GraphQL.SystemReactive/SubscriptionExecutionStrategy.cs

```diff
@@ -1,178 +0,0 @@
-        }
-
-        /// <summary>
-        /// Asynchronously returns a stream of <see cref="ExecutionResult"/> responses for the
-        /// specified <see cref="ExecutionNode"/>.
-        /// </summary>
-        protected virtual async Task<IObservable<ExecutionResult>?> ResolveEventStreamAsync(ExecutionContext context, ExecutionNode node)
-        {
-            context.CancellationToken.ThrowIfCancellationRequested();
-
-            object? source = (node.Parent != null)
-                ? node.Parent.Result
-                : context.RootValue;
-
-            try
-            {
-                var resolveContext = new ResolveFieldContext(new ReadonlyResolveFieldContext(node, context));
-
-                IObservable<object?> subscription;
-
-                if (node.FieldDefinition?.Subscriber != null)
-                {
-                    subscription = await node.FieldDefinition.Subscriber.SubscribeAsync(resolveContext).ConfigureAwait(false);
-                }
-                else
-                {
-                    throw new InvalidOperationException($"Subscriber not set for field '{node.Field.Name}'.");
-                }
-
-                // The IServiceProvider instance will be disposed at this point, set it to null instead of exposing the disposed object
-                resolveContext.RequestServices = null;
-
-                return subscription
-                    .Select(value => BuildSubscriptionExecutionNode(node.Parent!, node.GraphType!, node.Field, node.FieldDefinition, node.IndexInParentNode, value!))
-                    .SelectMany(async executionNode =>
-                    {
-                        if (context.Listeners != null)
-                        {
-                            foreach (var listener in context.Listeners)
-                            {
-                                await listener.BeforeExecutionAsync(context)
-                                    .ConfigureAwait(false);
-                            }
-                        }
-
-                        // Execute the whole execution tree and return the result
-                        await ExecuteNodeTreeAsync(context, executionNode).ConfigureAwait(false);
-
-                        if (context.Listeners != null)
-                        {
-                            foreach (var listener in context.Listeners)
-                            {
-                                await listener.AfterExecutionAsync(context)
-                                    .ConfigureAwait(false);
-                            }
-                        }
-
-                        // Set the execution node's value to null if necessary
-                        // Note: assumes that the subscription field is nullable, regardless of how it was defined
-                        // See https://github.com/graphql-dotnet/graphql-dotnet/pull/2240#discussion_r570631402
-                        //TODO: check if a non-null subscription field is allowed per the spec
```

Comment:
Allowed?

Reply:
Silent, so I figured it should be treated exactly the same way any other nullable or non-null field is treated.  Tests were added in `Widget_Null` and `Widget_Null_Nullable` to verify that the response data is correct, matching the same way if it were resolved as a query or mutation, with proper null bubbling for non-null fields.  If a user wants it to be nullable, they should define it as nullable.

The counter-argument is that the subscribed field is unknown since there is no map in the `data` response, since the null bubbled.  One would have to figure out from the error message (which does have a proper path), or keep track of it within the transport (which it does anyway).